File: backend/src/features/dashboard/schemas.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class DashboardSummary(BaseModel):
    income_total: Decimal
    expense_total: Decimal
    balance: Decimal


class CategorySpending(BaseModel):
    category_id: uuid.UUID
    category_name: str
    total: Decimal


class BalancePoint(BaseModel):
    date: date
    balance: Decimal


class DashboardRead(BaseModel):
    start_date: date
    end_date: date
    summary: DashboardSummary
    by_category: list[CategorySpending]
    balance_evolution: list[BalancePoint]

File: frontend/src/features/dashboard/schemas.ts
export type DashboardSummary = {
  income_total: string;
  expense_total: string;
  balance: string;
};

export type CategorySpending = {
  category_id: string;
  category_name: string;
  total: string;
};

export type BalancePoint = {
  date: string;
  balance: string;
};

export type Dashboard = {
  start_date: string;
  end_date: string;
  summary: DashboardSummary;
  by_category: CategorySpending[];
  balance_evolution: BalancePoint[];
};
